packages/next/src/build/page-info.ts

```diff
@@ -0,0 +1,57 @@
+import type { ServerRuntime } from '../../types'
+
+export interface PageInfo {
+  isHybridAmp?: boolean
+  size: number
+  totalSize: number
+  isStatic: boolean
+  isSSG: boolean
+  isPPR: boolean
+  ssgPageRoutes: string[] | null
+  initialRevalidateSeconds: number | false
+  pageDuration: number | undefined
+  ssgPageDurations: number[] | undefined
+  runtime: ServerRuntime
+  hasEmptyPrelude?: boolean
+  hasPostponed?: boolean
+  isDynamicAppRoute?: boolean
+}
+
+/**
+ * This class is used to store information about pages that is used for the
+ * build output.
+ */
+
+export class PageInfoRegistry extends Map<string, PageInfo> {
```

Comment:
extending a built-in in this way IMO obfuscates what the registry can do. For instance if I see the type of the registry is a map I know I can do Map-like stuff to it but if I see it is some class I have to build a new mental model of it's some base thing + some additional capabilities.

I'd argue that this particular abstraction isn't valuable and a simple `patchPageInfo` function would suffice everyone we call `registry.patch` and a `expectPageInfo()` function would suffice if you want to throw on missing page data. Then the other places where you interact with the page info as a regular Map are much more clear about what that object is